File: Codechef/HMAPPY2.cpp
#include<iostream>
#include<algorithm>
using namespace std;

typedef long long int ll;

ll gcd(ll a,ll b)
{
    if(a==0)
    {
        return b;
    }
    return gcd(b%a,a);
}

ll lcm(ll a,ll b)
{
    return a*b/gcd(a,b);
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    ll t,n,k,a,b;
    cin>>t;

    while(t--)
    {
        cin>>n>>a>>b>>k;
        ll num=n/a-n/lcm(a,b);
        num+=n/b-n/lcm(a,b);
        //cout<<num<<endl;
        if(num<k)
        {
            cout<<"Lose"<<endl;
        }
        else
        {
            cout<<"Win"<<endl;
        }
    }
}

File: Codechef/HMAPPY2.py
def gcd(a,b):
    if(a==0):
        return b
    else:
        return gcd(b%a,a)
    
def lcm(a,b):
    return a*b/gcd(a,b)

t=int(input())

for i in range(t):
    string=input()
    lst=string.split(" ")
    n=int(lst[0])
    a=int(lst[1])
    b=int(lst[2])
    k=int(lst[3])
    num=n/a+n/b-2*n/lcm(a,b)
    if num<k:
        print("Lose")
    else:
        print("Win")